src/main/logseq/api.cljs

```diff
@@ -477,11 +480,12 @@
                     before?)
           new-block (editor-handler/api-insert-new-block!
                       content
-                      {:block-uuid block-uuid'
-                       :sibling?   sibling?
-                       :before?    before?
-                       :page       page-name
-                       :properties properties})]
+                      {:block-uuid  block-uuid'
+                       :sibling?    sibling?
+                       :before?     before?
+                       :page        page-name
+                       :custom-uuid customUUID
+                       :properties  properties})]
```

Comment:
to be clear - I was hoping to insert a block with a specific `:block-uuid block-uuid'`. Is this not doable or even recommended bc of the reindexing? and that we should use the `customUUID` to reference a block? The downside of that approach is that it won't be indexed and will not be able to do `Editor.getBlock(customUUID)` or something similar as performantly

Reply:
Reindexing will reset it, but by using upsertBlockProperty and setting the property id, to the uuid, it will be persisted. 

Block UUID is actually the location for the block to be inserted under, i.e. if you want to insert it underneath another block, you will pass in the UUID of that block(the block under which you want the block to be inserted). Custom uuid will set the new block to have your customly defined UUID. 

Essentially it's doing exactly what you want it to do, it's just that block-uuid is referring to the source block while customUUID is referring to the uuid the new block will have.

Reply:
cool, I think so, thank you!